test/unit/BCoWPool.t.sol

```diff
@@ -137,3 +140,72 @@ contract BCoWPool_Unit_EnableTrading is BaseCoWPoolTest {
     bCoWPool.enableTrading(appData);
   }
 }
+
+contract BCoWPool_Unit_IsValidSignature is BaseCoWPoolTest {
+  function setUp() public virtual override {
+    super.setUp();
+    for (uint256 i = 0; i < TOKENS_AMOUNT; i++) {
+      vm.mockCall(tokens[i], abi.encodePacked(IERC20.approve.selector), abi.encode(true));
+    }
+    bCoWPool.finalize();
+  }
+
+  modifier _withValidCommitment(GPv2Order.Data memory _order) {
+    bCoWPool.enableTrading(_order.appData);
+    bytes32 _orderHash = GPv2Order.hash(_order, domainSeparator);
+    vm.prank(cowSolutionSettler);
+    bCoWPool.commit(_orderHash);
+    _;
+  }
```

Comment:
how can we cannot mock the commitment?
i understand that foundry tests are executed like in 1 tx, so everything that happens within a `test_Sth()...` scope, will have the transient storage unerased

Reply:
calling the `commit` method directly was the simplest option I found since

- the method itself is pretty simple
- I didn't find a way with `forge-std` to write to a particular _transient_ storage slot, I think it's a feature worth asking for/contributing

I'll add a setter with the same implementation as the `commit`, sans the checker, so changes in `commit` don't break this test